pkgs/development/tools/misc/ccache/default.nix

```diff
@@ -18,44 +18,46 @@ let ccache = stdenv.mkDerivation rec {
   doCheck = !stdenv.isDarwin;
 
   passthru = let
-      cc = stdenv.cc.cc;
-      ccname = if stdenv.cc.isClang then "clang" else "gcc";
-      cxxname = if stdenv.cc.isClang then "clang++" else "g++";
+      unwrappedCC = stdenv.cc.cc;
     in {
     # A derivation that provides gcc and g++ commands, but that
     # will end up calling ccache for the given cacheDir
     links = extraConfig: stdenv.mkDerivation rec {
       name = "ccache-links";
       passthru = {
-        inherit (stdenv.cc) isClang;
-        inherit (cc) isGNU;
+        isClang = unwrappedCC.isClang or false;
+        isGNU = unwrappedCC.isGNU or false;
       };
-      inherit (cc) lib;
+      inherit (unwrappedCC) lib;
```

Comment:
This must be for compatibility with other expressions:
https://github.com/NixOS/nixpkgs/blob/3f02cbbcaf18626e1f6a1824cfce142ad48ec614/pkgs/development/compilers/llvm/3.8/clang/default.nix#L44
https://github.com/NixOS/nixpkgs/blob/3f02cbbcaf18626e1f6a1824cfce142ad48ec614/pkgs/development/misc/amdapp-sdk/default.nix#L90